File: src/main/java/io/github/support/terminal/application/domains/updater/UpdaterController.java
package io.github.support.terminal.application.domains.updater;


import io.github.support.terminal.application.domains.updater.services.UpdaterService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class UpdaterController {

    private final UpdaterService updaterService;

    @PostMapping("/api/run-auto-update")
    public void runAutoUpdate() throws Exception {
        updaterService.runAutoUpdate();
    }

}

File: src/main/kotlin/io/github/support/terminal/application/domains/updater/UpdaterController.kt
package io.github.support.terminal.application.domains.updater

import io.github.support.terminal.application.domains.updater.services.UpdaterService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class UpdaterController(
    private val updaterService: UpdaterService
) {

    @PostMapping("/api/run-auto-update")
    @Throws(Exception::class)
    fun runAutoUpdate() {
        updaterService.runAutoUpdate()
    }
}
